### PR TITLE
Add serviceAnnotations helm value for all charts

### LINKED ISSUE #897: Provide an option to add annotation to vcluster services in vcluster-eks chart
### Is your feature request related to a problem?

We are using Consul on a cluster where multiple vclusters are installed using vcluster-eks helm chart. While enabling Consul service sync, all the vcluster services (api, etcd services, etc) are synced by default. One way to exclude them from service syncing is to add annotation on the service resources `'consul.hashicorp.com/service-sync': 'false'`. However, couldn't find a way to add such annotation through helm values.

### Which solution do you suggest?

Provide a helm value in vcluster-eks chart to add additional service annotations.

### Which alternative solutions exist?

In vcluster helm chart, there is an option `globalAnnotations`, could potentially work in our case.
We are also investigating config changes on the Consul side

### Additional context

*No response*

### PR DESCRIPTION
**Pull request by neogopher**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #897 

**Please provide a short message that should be published in the vcluster release notes**
Added a serviceAnnotations Helm value to all charts which can be used to set annotations for core vcluster services

Closes ENG-876
**What else do we need to know?** 
coredns component already has a helm value coredns.service.annotations which can be used for the same purpose

### CODE REVIEW

File: charts/eks/values.yaml [hunk 1]
```diff
@@ -238,11 +242,16 @@ api:
...
+  serviceAnnotations: {}
 
 # Core DNS settings
 coredns:
   image: public.ecr.aws/eks-distro/coredns/coredns:v1.8.7-eks-1-24-7
   replicas: 1
+  # CoreDNS service configurations
+  service:
+    # Extra Annotations
+    annotations: {}
```
Comment (matskiv) [inline charts/eks/values.yaml:254]: This has not been added to other charts. Let's keep them all in sync.

Comment (neogopher) [reply charts/eks/values.yaml:254]: This was already existing in other charts and was just missing in eks chart. So I added it here to keep all in sync.

File: charts/eks/templates/syncer-service.yaml [hunk 2]
```diff
@@ -8,9 +8,10 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
-  {{- if .Values.globalAnnotations }}
+  {{- $annotations := merge .Values.globalAnnotations .Values.syncer.serviceAnnotations }}
+  {{- if $annotations }}
   annotations:
-{{ toYaml .Values.globalAnnotations | indent 4 }}
+  {{ toYaml $annotations | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (neogopher) [reply charts/eks/templates/syncer-service.yaml:14]: Ah!! No, I think indent can be used here. I will make the changes.

**Review by FabianKramm — APPROVED** [review]

**Review by matskiv — APPROVED** [review]

File: charts/eks/templates/syncer-service.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (matskiv) [reply charts/eks/templates/syncer-service.yaml]: Thank you.

**Pull request merged**